### ISSUE TITLE
Configure verticalScrollMargin through Settings

### ISSUE THREAD

**Issue by hypnoglow** [atom/atom#11799]
**tldr**: [verticalScrollMargin](https://github.com/atom/atom/blob/73525a1bcc309ff64c6f930401cedc052ddcf408/src/text-editor.coffee#L71) should be configurable through Settings (`config.cson`).

I had a hard time searching for vim-like [scrolloff](http://vim.wikia.com/wiki/Make_search_results_appear_in_the_middle_of_the_screen#Scrolloff_option) feature in Atom.

Finally, I figured out that for this you have to write something like this in your **init.coffee**:

```
atom.workspace.observeTextEditors (editor) =>
    editor.setVerticalScrollMargin 8
```

There is even a [package](https://atom.io/packages/atom-scrolloff) where they tried to implement this feature because users experience difficulties trying to configure this.
